runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/testing/TestDataflowRunner.java

```diff
@@ -313,6 +314,8 @@ public String toString() {
     private final DataflowPipelineJob job;
     private final JobMessagesHandler messageHandler;
     private final StringBuffer errorMessage;
+    private final AtomicBoolean haveErrors = new AtomicBoolean();
```

Comment:
I would probably just call this `cancelled`. That describes at least more accurately how it would be used (guarding the actual call to cancel), rather than the situation that caused the job to be cancelled.